### PR TITLE
DMP-963: Save last search

### PR DESCRIPTION
**Pull request by deanbullock-hmcts**
### Jira link (if applicable)

https://tools.hmcts.net/jira/browse/DMP-963

### CODE REVIEW

File: cypress/e2e/functional/case-search-spec.cy.js [hunk 1]
```diff
@@ -149,6 +149,48 @@ describe('Case search', () => {
...
+    // Check case file page
+    cy.get('h1').should('contain', 'C20220620001');
+
+    // << Go Back
+    cy.get('.govuk-back-link').click();
+
+    // Check for Search page
+    cy.get('h1').should('contain', 'Search for a case');
+
+    cy.contains('Advanced search').click();
```
Comment (davet1985) [inline cypress/e2e/functional/case-search-spec.cy.js:183]: I think advanced search should be open if any of the form values are populated

File: src/app/components/common/courthouse/courthouse.component.ts [hunk 2]
```diff
@@ -48,6 +49,7 @@ export class CourthouseComponent implements AfterViewInit, OnChanges {
     if (this.courthouses.length) {
       this.props.element = this.autocompleteContainer.nativeElement as HTMLElement;
       this.props.source = this.courthouses.map((ch) => ch.courthouse_name);
+      this.props.defaultValue = this.courthouse;
```
Comment (davet1985) [inline src/app/components/common/courthouse/courthouse.component.ts:52]: Can you add a unit test for this?

File: src/app/components/search/search-error/search-error.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+      imports: [SearchErrorComponent],
+    });
+    fixture = TestBed.createComponent(SearchErrorComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
```
Comment (davet1985) [inline src/app/components/search/search-error/search-error.component.spec.ts:20]: Maybe some more tests here to check the content for each error type?

File: src/app/components/search/search-error/search-error.component.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { Component, Input } from '@angular/core';
+import { CommonModule } from '@angular/common';
+
+@Component({
+  selector: 'app-search-error',
+  standalone: true,
+  imports: [CommonModule],
```
Comment (davet1985) [inline src/app/components/search/search-error/search-error.component.ts:7]: Would it be better to just import `NgSwitch`?

**Review by davet1985 — COMMENTED** [review]
Looks good, some minor comments.

File: cypress/e2e/functional/case-search-spec.cy.js (hunk 1 above)
Comment (deanbullock-hmcts) [reply cypress/e2e/functional/case-search-spec.cy.js:183]: Agreed 👍

File: src/app/components/common/courthouse/courthouse.component.ts (hunk 2 above)
Comment (deanbullock-hmcts) [reply src/app/components/common/courthouse/courthouse.component.ts:52]: Yes. I need to take a look at the fallback too actually. In the event of no courthouses returned.

File: src/app/components/search/search.component.ts [hunk 5]
```diff
@@ -41,25 +41,26 @@ const FieldErrors: { [key: string]: { [key: string]: string } } = {
...
   courthouses$ = this.caseService.getCourthouses();
+  courthouse = '';
+
+  // Retrieve Previous Search Results
+  searchResults$ = this.caseService.searchResults$;
+  searchError$: Observable<string> | null = null;
   subs: Subscription[] = [];
 
-  constructor(private caseService: CaseService) {}
+  constructor(public caseService: CaseService) {}
```
Comment (davet1985) [inline src/app/components/search/search.component.ts:63]: Why is this public now?

Comment (deanbullock-hmcts) [reply src/app/components/search/search.component.ts:63]: It shouldn't be, I set it to public to access it from template but then didn't need to. I will revert.

**Review by davet1985 — APPROVED** [review]

**Pull request merged**